libs/components/src/progress-bar/vl-progress-bar.component.cy.ts

```diff
@@ -0,0 +1,93 @@
+import { registerWebComponents } from '@domg-wc/common-utilities';
+import { VlTooltipComponent } from '../tooltip/vl-tooltip.component';
+import { VlProgressBarComponent } from './vl-progress-bar.component';
+import { html } from 'lit';
+
+registerWebComponents([VlProgressBarComponent, VlTooltipComponent]);
+
+const mountDefault = (steps: string[]) => cy.mount(html`<vl-progress-bar .steps=${steps}></vl-progress-bar> `);
+
+describe('component vl-progress-bar - default', () => {
+    const steps = ['Step 1', 'Step 2', 'Step 3'];
+
+    beforeEach(() => {
+        mountDefault(steps);
+    });
+
+    it('should mount', () => {
+        cy.get('vl-progress-bar');
+    });
+
+    it('should be accessible', () => {
+        cy.injectAxe();
+        cy.checkA11y('vl-progress-bar');
+    });
+
+    it('should render steps correctly', () => {
+        cy.get('vl-progress-bar').shadow().find('.vl-progress-bar__step').should('have.length', steps.length);
+
+        steps.forEach((step, index) => {
+            cy.get('vl-progress-bar')
+                .shadow()
+                .find(`.vl-progress-bar__step:nth-child(${index + 1}) `)
+                .find('vl-tooltip')
+                .contains(step);
+        });
+    });
+
+    it('should emit vl-click-step event when a step is clicked', () => {
+        cy.get('vl-progress-bar').shadow().find('.vl-progress-bar__step:nth-child(2) button').click();
+
+        cy.get('vl-progress-bar').then(($el) => {
+            const el = $el[0];
+            const event = new CustomEvent('vl-click-step', {
+                bubbles: true,
+                composed: true,
+                detail: { step: steps[1], number: 2 },
+            });
+
+            expect(el.dispatchEvent(event)).to.be.true;
+        });
+    });
+});
+
+describe('component vl-progress-bar - with properties', () => {
+    const mountWithProps = (steps: string[], numeric: boolean, activeStep: number, alwaysShowSteps: boolean) =>
+        cy.mount(
+            html`<vl-progress-bar
+                .steps=${steps}
+                .numeric=${numeric}
+                .activeStep=${activeStep}
+                .alwaysShowSteps=${alwaysShowSteps}
+            ></vl-progress-bar>`
+        );
+
+    const steps = ['Step 1', 'Step 2', 'Step 3'];
+    const numeric = true;
+    const activeStep = 2;
+    const alwaysShowSteps = true;
+
+    beforeEach(() => {
+        mountWithProps(steps, numeric, activeStep, alwaysShowSteps);
```

Comment:
Als je dit doet voor elke story is het niet nodig om dit in een aparte methode te steken.
Je mag deze props hardcoded meegeven in de cy.mount() call.

Reply:
gedaan.